dumpling/export/dump.go

```diff
@@ -351,7 +362,17 @@ func (d *Dumper) dumpDatabases(tctx *tcontext.Context, metaConn *sql.Conn, taskC
 			if err != nil {
 				return err
 			}
-
+			// adjust table collation
+			if table.Type == TableTypeBase {
+				newCreateSQL, defaultCollation, err := adjustTableCollation(metaConn, parser1, meta.ShowCreateTable(), dbCollation, dbName, table.Name)
+				if err != nil {
+					return err
+				}
+				if dbCollation == "" {
```

Comment:
seems this should move out of for loop?

Reply:
it is used to be database collation and will be use table collation when table collation is Implicit. if it is not inited before, it will be inited when adjust the first table.